src/lib/structures/SlashCommandStore.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * This is the most basic of examples for a store, this is all you really need
+ * to actively register a store with Sapphire, and anything in the supplied
+ * folder will load if it extends a Sapphire piece.
+ */
+const SlashCommand = require("./SlashCommandPiece.js");
+const { Store } = require("@sapphire/framework");
+
+module.exports = class SlashCommandStore extends Store {
+  constructor() {
+    // This is the name of the directory we want to look in for our slash
+    // commands.
+    super(SlashCommand, { name: "slashCommands" });
+  }
+
+  async registerCommands() {
+    const client = this.container.client;
+    if(!client) return;
+
+    // This will split the slash commands between global and guild only.
+    const slashCommands = this.container.stores.get('slashCommands');
+    const [guildCmds, globalCmds] = slashCommands.partition(c => c.guildOnly);
+    
+    // iterate to all connected guilds and apply the commands.
+    const guilds = await client?.guilds?.fetch(); // retrieves Snowflake & Oauth2Guilds
+    for(let [id] of guilds) {
+      const guild = await client?.guilds?.fetch(id); // gets the guild instances from the cache (fetched before)
```

Comment:
Since the fetch from `guilds` returns an array of arrays with `id` and `Oauth2Guild` we need to fetch the actual `Guild` instance. This will be grabbed from cache.